### PR TITLE
Fix 'drawAllInList' function

### PR DESCRIPTION
**Pull request by TakoTheDev**
At least in the demo renderer it seems to not align correctly when any scaling is applied (Also fixes 3 minor issues: Dungeon Timer height is incorrect, the width and demo renderer for Giant Names don't use same length and the dungeon profit width is missing a +)

**BEFORE**
image

**AFTER**
image
(idk how to fix that very slight space after the heart of the Giant display)

### CODE REVIEW

File: src/main/kotlin/gg/skytils/skytilsmod/utils/RenderUtil.kt [hunk 1]
```diff
@@ -863,10 +863,11 @@ object RenderUtil {
     }
 
     fun drawAllInList(element: GuiElement, lines: Collection<String>) {
+        val width = lines.maxOf { ScreenRenderer.fontRenderer.getStringWidth(it) }
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/utils/RenderUtil.kt]: Is there a reason why this is used rather than the scaled width? It doesn't seem to take scaling into account.

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/utils/RenderUtil.kt]: Imma commit an improvement in a second, for some reason `element.width` (WIDTHWIDTH) is the correct width, `element.scaleWidth` (SCALEDWIDTH) is wrong and can't be bothered to find out why. (CORRECTWIDTH is the string width)
image

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/utils/RenderUtil.kt]: So tldr: element width is used instead of scaled width since it's wrong for some reason

**Comment by TakoTheDev** [comment]
Also as a side note, optifine seems to break the width calculations?
image
image
(Counts for both old and new editing gui, but can probably not be fixed easily so I'll just leave it like that)

File: src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt [hunk 2]
```diff
@@ -176,6 +179,10 @@ object DungeonListener {
                         val partyCount = partyCountPattern.find(tabEntries[0].second)?.groupValues?.get(1)?.toIntOrNull()
                         if (partyCount != null) {
                             println("There are $partyCount members in this party")
+                            if (team.size != partyCount) {
+                                println("Clearing team as the party size has changed ${team.size} -> $partyCount")
+                                team.clear()
+                            }
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt]: Why is all this changed? It seems unrelated

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/ChestProfit.kt [hunk 3]
```diff
@@ -312,7 +312,7 @@ object ChestProfit {
         override val height: Int
             get() = ScreenRenderer.fontRenderer.FONT_HEIGHT * DungeonChest.entries.size
         override val width: Int
-            get() = ScreenRenderer.fontRenderer.getStringWidth("Obsidian Chest: 300M")
+            get() = ScreenRenderer.fontRenderer.getStringWidth("Obsidian Chest: §a+300M")
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/ChestProfit.kt]:
```suggestion
            get() = ScreenRenderer.fontRenderer.getStringWidth("Obsidian Chest: +300M")
```

**Review by Sychic — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt (hunk 2 above)
Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt]: this is caused by me not merging the latest commit from the branch I think xd

**Comment by TakoTheDev** [comment]
I forgot to add [no ci] ;-;

**Review by Sychic — CHANGES_REQUESTED** [review]
Remove the random newlines

File: src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt:1]: This one too

**Review by Sychic — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt
Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/listeners/DungeonListener.kt:1]: I can't push the random space after the } cause git or intellij ;-;

**Comment by TakoTheDev** [comment]
It had a new line but without the \r\n thing ig?

**Review by My-Name-Is-Jeff — APPROVED** [review]
